### PR TITLE
feat: elk code on dco fixed branch

### PR DESCRIPTION
**Pull request by tipusinghaw**
#### What
- Implemented ELK logging on all the platform services
- Added and configured Winston for logger

### CODE REVIEW

**Review by RinkalBhojani — APPROVED** [review]
Just check sonar issues and if possible resolve it. Rest all LGTM

File: libs/config/src/config.module.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  imports: [
+    NestConfigModule.forRoot({
+      isGlobal: true
+    })
+  ],
+  controllers: [],
+  providers: [ConfigService],
+  exports: [ConfigService]
+})
+export class ConfigModule {}
```
Comment (GHkrishna) [inline libs/config/src/config.module.ts:16]: We can maybe have all the exports in the respective `index.ts`

File: libs/config/src/config.service.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService as NestConfigService } from '@nestjs/config';
+
+@Injectable()
+export class ConfigService {
```
Comment (GHkrishna) [inline libs/config/src/config.service.ts:5]: We can maybe have all the exports in the respective `index.ts`

File: libs/config/src/global-config.module.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+@Module({
+  providers: [
+    {
+      provide: MICRO_SERVICE_NAME,
+      useValue: 'DefaultServiceName'
+    }
+  ],
+  exports: [MICRO_SERVICE_NAME]
+})
+export class GlobalConfigModule {}
```
Comment (GHkrishna) [inline libs/config/src/global-config.module.ts:14]: We can maybe have all the exports in the respective `index.ts`

File: apps/api-gateway/src/agent-service/agent-service.module.ts [hunk 4]
```diff
@@ -8,6 +8,7 @@ import { AgentController } from './agent-service.controller';
 import { AgentService } from './agent-service.service';
 import { getNatsOptions } from '@credebl/common/nats.config';
 import { CommonConstants } from '@credebl/common/common.constant';
+import { NATSClient } from 'libs/common/NATSClient';
```
Comment (GHkrishna) [inline apps/api-gateway/src/agent-service/agent-service.module.ts:11]:
```suggestion
import { NATSClient } from '@credebl/common/NATSClient';
```

File: apps/verification/src/verification.service.ts [hunk 5]
```diff
@@ -118,9 +121,9 @@ export class VerificationService {
     const payload = {
       templateIds
     };
-    const schemaAndOrgDetails = await this.verificationServiceProxy
-      .send(pattern, payload)
-      .toPromise()
+    const schemaAndOrgDetails = await this.natsClient
+      .send<ISchemaDetail[]>(this.verificationServiceProxy, pattern, payload)
+      // .toPromise()
```
Comment (GHkrishna) [inline apps/verification/src/verification.service.ts:126]: Can we remove this?

File: apps/issuance/src/issuance.service.ts [hunk 6]
```diff
@@ -1299,9 +1304,9 @@ async sendEmailForCredentialOffer(sendEmailCredentialOffer: SendEmailCredentialO
     const payload = {
       templateIds
     };
-    const schemaDetails = await this.issuanceServiceProxy
-      .send(pattern, payload)
-      .toPromise()
+    const schemaDetails = await this.natsClient
+      .send<ISchemaDetail[]>(this.issuanceServiceProxy, pattern, payload)
+      // .toPromise()
```
Comment (GHkrishna) [inline apps/issuance/src/issuance.service.ts:1309]:
```suggestion

```

File: apps/organization/src/organization.service.ts [hunk 7]
```diff
@@ -557,9 +560,9 @@ export class OrganizationService {
         orgId
       }
     };
-    const connectionInvitationData = await this.organizationServiceProxy
-      .send(pattern, payload)
-      .toPromise()
+    const connectionInvitationData = await this.natsClient
+      .send<ICreateConnectionUrl>(this.organizationServiceProxy, pattern, payload)
+      // .toPromise()
```
Comment (GHkrishna) [inline apps/organization/src/organization.service.ts:565]:
```suggestion

```

File: libs/logger/src/winstonLogger.ts [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    });
+
+    return {
+      level: LogLevel.Debug,
+      levels,
+     // format: ecsFormat.ecsFormat({ convertReqRes: true }),
+       format: winston.format.combine(
+        ecsFormat.ecsFormat({ convertReqRes: true }),
+        // Add timestamp and format the date
+        // winston.format.timestamp({
```
Comment (GHkrishna) [inline libs/logger/src/winstonLogger.ts:62]: Do we need this commented code

File: libs/config/src/config.service.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(private readonly configService: NestConfigService) {}
+
+  get isProduction(): boolean {
+    return 'production' === this.environment;
```
[the remark this answers is not in the document]
Comment (tipusinghaw) [reply libs/config/src/config.service.ts:9]: Resolve

**Review by GHkrishna — APPROVED** [review]

**Review by GHkrishna — APPROVED** [review]

**Pull request merged**